packages/valory/skills/price_estimation_abci/models.py

```diff
@@ -77,8 +80,56 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         """Initialize PriceApi."""
         self.convert_id = self.ensure("convert_id", kwargs)
         self.currency_id = self.ensure("currency_id", kwargs)
+        self.response_index: Optional[int] = kwargs.pop("response_index", None)
         super().__init__(*args, **kwargs)
 
+    def _log_response(self, decoded_response: str) -> None:
+        """Log the decoded response message using error level."""
+        self.context.logger.error(f"\nResponse: {decoded_response}")
+
+    def _get_value_with_type(self, value: Any) -> None:
+        """Get the given value as the specified type."""
+        return getattr(builtins, self.response_type)(value)
+
+    def _get_response_from_index(self, value: List[Any]) -> None:
+        """Get the response using the given index."""
+        if self.response_index is not None:
+            value = value[self.response_index]
+        return self._get_value_with_type(value)
+
+    def _get_response_data(
+        self, response: HttpMessage, response_key: Optional[str], _response_type: str
+    ) -> Any:
+        """Get response data from api, based on the given response key"""
+        decoded_response = response.body.decode()
+
+        try:
+            response_data = json.loads(decoded_response)
+            if response_key is None:
+                return self._get_response_from_index(response_data)
+
+            first_key, *keys = response_key.split(":")
+            value = response_data[first_key]
+            for key in keys:
+                value = value[key]
+
+            return self._get_response_from_index(value)
+
+        except json.JSONDecodeError:
+            self.context.logger.error("Could not parse the response body!")
+        except KeyError:
+            self.context.logger.error(
+                f"Could not access response using the given key(s) ({self.response_key})!"
+            )
+        except IndexError:
+            self.context.logger.error(
+                f"Could not access response using the given key(s) ({self.response_key}) "
+                f"and index ({self.response_index})!"
+            )
+
+        self._log_response(decoded_response)
+        return None
```

Comment:
It would be better if we moved this to `open-autonomy` so that all future services can utilize it. Moreover, it would remove the HTTP protocol dependency from the `price_estimation_abci` skill:

https://github.com/valory-xyz/price-oracle/blob/18a86ef8014573eace8cef47d44afb3fda543b28/packages/valory/skills/price_estimation_abci/skill.yaml#L37

which is only necessary because of `HttpMessage`:

https://github.com/valory-xyz/price-oracle/blob/18a86ef8014573eace8cef47d44afb3fda543b28/packages/valory/skills/price_estimation_abci/models.py#L101

Reply:
Yes please do this asap for the next release